### PR TITLE
expression: fix wrong flen infer for bit constant

### LINKED ISSUE #23479: the behavior of bit type for / is incompatible with MySQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
select b'10000000' DIV 10;
select b'10000000' / 10;
select cast(b'10000000' as unsigned) / 10;
```

### 2. What did you expect to see? (Required)
```sql
mysql > select b'10000000' / 10;
+------------------+
| b'10000000' / 10 |
+------------------+
| 12.8000          |
+------------------+


```
### 3. What did you see instead (Required)
```sql
mysql > select b'10000000' DIV 10;
+--------------------+
| b'10000000' DIV 10 |
+--------------------+
| 12                 |
+--------------------+

mysql > select cast(b'10000000' as unsigned) / 10;
+------------------------------------+
| cast(b'10000000' as unsigned) / 10 |
+------------------------------------+
| 12.8000                            |
+------------------------------------+

mysql > select b'10000000' / 10;
+------------------+
| b'10000000' / 10 |
+------------------+
| 0.9000           |
+------------------+

```
### 4. What is your TiDB version? (Required)
master: fd706ab76bd0
release-5.0: 03f11d24c7

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Issue Number: close #23479

### What is changed and how it works?

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- fix wrong flen infer for bit constant

### CODE REVIEW

File: types/field_type.go [hunk 1]
```diff
@@ -250,7 +250,7 @@ func DefaultTypeForValue(value interface{}, tp *FieldType, char string, collate
 		SetBinChsClnFlag(tp)
 	case BitLiteral:
 		tp.Tp = mysql.TypeVarString
-		tp.Flen = len(x)
+		tp.Flen = len(x) * 3
```
Comment (XuHuaiyu) [inline types/field_type.go:253]: Why make this change?

Comment (lzmhhh123) [reply types/field_type.go:253]: Because the max bit value for a byte is `127` which flen is 3

**Review by ichn-hu — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

**Comment by hi-rustin** [comment]
/merge cancel 

for consistent failures.

@lzmhhh123 Could you please take a look?

**Pull request merged**